src/clims/services/extensible.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import
+
+from clims.services.substance import ExtensibleBaseField
+from clims.models import ExtensibleType, ExtensiblePropertyType
+from django.db import transaction
+
+
+class ExtensibleService(object):
+    def register(self, plugin, extensible_base):
```

Comment:
I'm aware that this is a lot to take in, since we have the whole model story and then the higher-level abstractions on top. I hope that we can name them more clearly with time.